tests/test-cases/Group23-VIC-Machine-Service/23-06-VCH-Certificate.robot

```diff
@@ -0,0 +1,103 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+*** Settings ***
+Documentation  Test 23-06 - VCH Certificate
+Resource  ../../resources/Util.robot
+Suite Setup  Start VIC Machine Server
+Suite Teardown  Terminate All Processes  kill=True
+Default Tags
+
+*** Keywords ***
+Install VIC Machine Without TLS
+    [Arguments]  ${vic-machine}=bin/vic-machine-linux  ${appliance-iso}=bin/appliance.iso  ${bootstrap-iso}=bin/bootstrap.iso  ${certs}=${true}  ${vol}=default  ${cleanup}=${true}  ${debug}=1  ${additional-args}=${EMPTY}
+    Set Test Environment Variables
+    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run  govc host.esxcli network firewall set -e false
+    # Attempt to cleanup old/canceled tests
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Dangling VMs On Test Server
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Datastore On Test Server
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Dangling Networks On Test Server
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Dangling vSwitches On Test Server
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Dangling Containers On Test Server
+    Run Keyword If  ${cleanup}  Run Keyword And Ignore Error  Cleanup Dangling Resource Pools On Test Server
+    Set Suite Variable  ${vicmachinetls}  --no-tls
+    Log To Console  \nInstalling VCH to test server with tls disabled...
+    ${output}=  Run VIC Machine Command  ${vic-machine}  ${appliance-iso}  ${bootstrap-iso}  ${certs}  ${vol}  ${debug}  ${additional-args}
+    Log  ${output}
+    Should Contain  ${output}  Installer completed successfully
+
+    Get Docker Params  ${output}  ${certs}
+    Log To Console  Installer completed successfully: %{VCH-NAME}...
+
+    [Return]  ${output}
+
+Start VIC Machine Server
+    Start Process  ./bin/vic-machine-server --port 31337  shell=True  cwd=/go/src/github.com/vmware/vic
+
+Curl No Datacenter
+    [Arguments]  ${vch-id}  ${auth}
+    ${rc}  ${output}=  Run And Return Rc And Output  curl -s -w "\%{http_code}\n" -X GET "http://127.0.0.1:31337/container/target/%{TEST_URL}/vch/${vch-id}/certificate?thumbprint=%{TEST_THUMBPRINT}" -H "authorization: Basic ${auth}"
+    [Return]  ${rc}  ${output}
+
+Curl Datacenter
+    [Arguments]  ${vch-id}  ${auth}
+    ${orig}=  Get Environment Variable  TEST_DATACENTER
+    ${dc}=  Run Keyword If  '%{TEST_DATACENTER}' == '${SPACE}'  Get Datacenter Name
+    Run Keyword If  '%{TEST_DATACENTER}' == '${SPACE}'  Set Environment Variable  TEST_DATACENTER  ${dc}
+    ${rc}  ${output}=  Run And Return Rc And Output  curl -s -w "\%{http_code}\n" -X GET "http://127.0.0.1:31337/container/target/%{TEST_URL}/datacenter/%{TEST_DATACENTER}/vch/${vch-id}/certificate?thumbprint=%{TEST_THUMBPRINT}" -H "authorization: Basic ${auth}"
```

Comment:
It looks like one of my recent commits broke this; I fixed the handler code to expect a datacenter id instead of name.

I don't know the idiomatic way to do this in robot, but: you can convert the name to the id using something like `govc datacenter.info -k --json -dc $TEST_DATACENTER | jq .Datacenters[0].Self.Value`.

Reply:
I added a new keyword `Get Datacenter ID` in `Vsphere-Util.robot` to handle this.